elements/classpower.lua

```diff
@@ -56,6 +62,32 @@ local function UpdateTexture(element)
 	end
 end
 
+local function UpdateStatusBar(element, full, partial, max)
+	for i = 1, max do
+		element[i]:Show()
+
+		if(i <= math.ceil(full + partial)) then
+			if(i <= full) then
+				element[i]:SetValue(1)
+			else
+				element[i]:SetValue(partial)
+			end
+		else
+			element[i]:SetValue(0)
+		end
+	end
+end
```

Comment:
Because it'd look sloppy. I (and to be fair not only I) use "smooth" status bars, I want gain and loss animations/processes to be consistent in both cases.